src/cpptools.ts

```diff
@@ -38,51 +42,109 @@ interface TargetDefaults {
 }
 
 function parseCppStandard(std: string): StandardVersion|null {
+  const is_gnu = std.startsWith('gnu');
```

Comment:
Before we send back the new modes, we need to check that the config provider is v4 or higher. If it's not, we need to revert to the old behavior.  You can query the api when you get it from cpptools.  Speaking of which, it looks like you need to update `extension.ts` line 473 to ask for v4 support.